From b192dda1d4b0f853a9fa01ac8e1b461c8d535a8e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 01 Apr 2022 09:17:36 +0300
Subject: [PATCH] [SW-77251] gaudi3: update recovery progress bit

This patch adds indication of ROM and EROM recovery as part of
arcboot status update.

Change-Id: I09a5da62dc4a9221406a3b087a2befbea9d551aa
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 51cca2e..9161456 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -40,6 +40,7 @@
 
 static void execute_erom_recovery(void)
 {
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_EROM_RECOVERY);
 	/* TODO: implement erom recovery */
 }
 
@@ -74,6 +75,8 @@
 
 static void execute_rom_recovery(void)
 {
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_ROM_RECOVERY);
+
 	rom_pcie_recovery();
 	rom_recovery();
 }
diff --git a/include/arcboot.h b/include/arcboot.h
index ec6045b..3133c8e 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -178,39 +178,41 @@
 	PROGRESS_ARCBOOT_RAM_START = 60,
 	PROGRESS_ARCBOOT_RECOVERY = 61,
 	PROGRESS_ARCBOOT_RECOVERY_DMA = 62,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_START = 63,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_DEST = 64,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 65,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_ERROR = 66,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_FINISH = 67,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 68,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_VERIFICATION_FAIL = 69,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR = 70,
-	PROGRESS_ARCBOOT_EROM_PCIE = 71,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 72,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 73,
-	PROGRESS_ARCBOOT_EROM_PCIE_START = 74,
-	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 75,
-	PROGRESS_ARCBOOT_EROM_RECOVERY = 76,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 77,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 78,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 79,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 80,
-	PROGRESS_ARCBOOT_EROM_TPM = 81,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 82,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 83,
-	PROGRESS_ARCBOOT_EROM_TPM_START = 84,
-	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 85,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_UP = 86,
-	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_HEADER_CHECK_FAILED = 87,
-	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_NO_DESCRIPTION = 88,
-	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_NO_IMAGES_PARENT_NODE = 89,
-	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_CORRUPTED_LOAD_TYPE = 90,
-	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_UNSUPPORTED_LOAD_TYPE = 91,
-	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_UNKNOWN_ERROR = 92,
-	PROGRESS_ARCBOOT_ROM_FIT_LOAD_FAILED = 93,
-	PROGRESS_ARCBOOT_ROM_FIT_LOAD_SEC_FAILED = 94,
-	PROGRESS_ARCBOOT_ROM_FIT_LOAD_UNKNOWN_ERROR = 95,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 63,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_START = 64,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_DEST = 65,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 66,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_ERROR = 67,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_FINISH = 68,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 69,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_VERIFICATION_FAIL = 70,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR = 71,
+	PROGRESS_ARCBOOT_EROM_PCIE = 72,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 73,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 74,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 75,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 76,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 77,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 78,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 79,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 80,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 81,
+	PROGRESS_ARCBOOT_EROM_TPM = 82,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 83,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 84,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 85,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 86,
+	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 87,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_UP = 88,
+	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_HEADER_CHECK_FAILED = 89,
+	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_NO_DESCRIPTION = 90,
+	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_NO_IMAGES_PARENT_NODE = 91,
+	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_CORRUPTED_LOAD_TYPE = 92,
+	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_UNSUPPORTED_LOAD_TYPE = 93,
+	PROGRESS_ARCBOOT_ROM_FIT_FORMAT_UNKNOWN_ERROR = 94,
+	PROGRESS_ARCBOOT_ROM_FIT_LOAD_FAILED = 95,
+	PROGRESS_ARCBOOT_ROM_FIT_LOAD_SEC_FAILED = 96,
+	PROGRESS_ARCBOOT_ROM_FIT_LOAD_UNKNOWN_ERROR = 97,
 };
 
 enum security_location {
